pkg/bpf/ops_linux.go

```diff
@@ -46,22 +46,17 @@ func (m StructBinaryMarshaler) MarshalBinary() ([]byte, error) {
 }
 
 type mapOps[KV KeyValue] struct {
-	m *ebpf.Map
-}
-
-func NewMapOps2[KV KeyValue](m *ebpf.Map) reconciler.Operations[KV] {
-	ops := &mapOps[KV]{m}
-	return ops
+	m *Map
 }
 
 func NewMapOps[KV KeyValue](m *Map) reconciler.Operations[KV] {
-	ops := &mapOps[KV]{m.m}
+	ops := &mapOps[KV]{m}
```

Comment:
TODO move these changes into their own commit
Issue was that the `m.m` was only set when `OpenOrCreate` was called and that happened from the start hook. So we cannot use it yet here, but need to postpone the dereference to later when the app has started.